### PR TITLE
Deck page updates

### PR DESCRIPTION
**Pull request by lily-pytel**
This PR got a lot more ridiculous than first intended. Phew. It's definitely not my best work since it's so rushed, but... it's a thing that does stuff. Hopefully it passes inspections. Full disclosure: I did not run the cypress tests. They might be broken now. SORRY!

Deck builder page:

image

Deck page:

image

### CODE REVIEW

File: express/index.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ app.use('/graphql', async (req, res, next) => {
     return acc && allowed;
   }, true);
   if (!allQueriesOk) {
-    res.status(500).send('Nope nope nope...');
+    res.status(500).send('This query is not allowed.');
```
Comment (jtrussell) [inline express/index.js]: Aww... ok.

File: next/components/deck-card-table.js [hunk 2]
```diff
@@ -35,40 +37,51 @@ export default function DeckCardsTable({ deck, deleteCard }) {
...
+                </td>
+                <td style={{ backgroundColor, color }}>{deckCard.card.name}</td>
+                <td>x{deckCard.quantity}</td>
+                {deleteCard && (
+                  <td
+                    data-cy="deckDeleteCard"
+                    className="deck-delete-card"
+                    onClick={() => deleteCard(deckCard.card.id)}
+                  >
+                    x
```
Comment (jtrussell) [inline next/components/deck-card-table.js]: `&times;` for life!

File: next/components/deck-delete.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ function DeckDelete({ deck }) {
   });
 
   // Users can only delete decks they authored
-  if (!user || user.id !== deck.author.id) return null;
+  if (!user || !deck || !deck.author || user.id !== deck.author.id) return null;
```
Comment (jtrussell) [inline next/components/deck-delete.js]: Respect the 🎩

File: next/lib/card.js [hunk 4]
```diff
@@ -29,3 +30,39 @@ export const imagePathSmall = (name, set = 'core') => {
...
+    return card.cardFactions.nodes[0].faction.name;
+  } catch (e) {
+    return null;
+  }
+};
+
+export const cardMainColor = (card, theme) => {
+  const faction = mainFaction(card);
+
+  let color = null;
```
Comment (jtrussell) [inline next/lib/card.js]: If you don't enjoy switches, here's another pattern I enjoy:

```
const color = {
  [FACTION_COLORS.blue]: theme.blueFactionColor,
  [FACTION_COLORS.yellow]: theme.yellowFactionColor,
  // etc.
}[faction] || 'default color'
```

Actually, now that I say it I can't remember if the babel configs allow for `{ [foo]: bar }` syntax 🤔.

File: express/index.js (hunk 1 above)
Comment (lily-pytel) [reply express/index.js]: Can be put back but I was a bit confused about it. For my own future amnesia, I considered making the error "did you remember to whitelist and restart Express?" But that seems like giving the hackers a lot of info lol

File: next/components/deck-card-table.js (hunk 2 above)
Comment (lily-pytel) [reply next/components/deck-card-table.js]: Do it

File: next/components/deck-delete.js (hunk 3 above)
Comment (lily-pytel) [reply next/components/deck-delete.js]: Got a page of death on one of our test decks that has no author :mmhmm:

File: express/index.js (hunk 1 above)
Comment (jtrussell) [reply express/index.js]: No. Your message is better, the more info the better short of handing out secrets.

**Pull request merged**